src/components/PadBtn/PadBtn.tsx

```diff
@@ -76,17 +191,78 @@ const PadBtn: React.FC<Props> = ({ input, calc, setCalc }) => {
       case "÷":
         a /= b;
         break;
-      default: //undefined
+      case "^":
+        a = a ** b;
+        break;
+      case "√":
+      case "π":
+        a = a;
+        break;
+      default: //should i remove this?
         a = b;
         break;
     }
     return a;
   };
 
+  const handleExponetiation = (signIndex: number, newCalc: Calc) => {
+    let equationCalc = { ...newCalc };
+    if (signIndex !== -1 && equationCalc.nums[signIndex + 1]) {
+      equationCalc.nums[signIndex + 1] = getResult(
+        parseFloat(equationCalc.nums[signIndex]),
+        parseFloat(equationCalc.nums[signIndex + 1]),
+        newCalc.sign[signIndex]
+      ).toString();
+
+      equationCalc.nums.splice(signIndex, 1);
+      equationCalc.sign.splice(signIndex, 1);
+
+      signIndex = equationCalc.sign.findIndex((element) => element === "^");
+      if (signIndex === -1) return;
+      handleMultiplicationsAndDivisions(signIndex, equationCalc);
+    }
+  };
+
+  const handleMultiplicationsAndDivisions = (
+    signIndex: number,
+    newCalc: Calc
+  ) => {
+    let equationCalc = { ...newCalc };
+    if (signIndex !== -1 && equationCalc.nums[signIndex + 1]) {
+      equationCalc.nums[signIndex + 1] = getResult(
+        parseFloat(equationCalc.nums[signIndex]),
+        parseFloat(equationCalc.nums[signIndex + 1]),
+        newCalc.sign[signIndex]
+      ).toString();
+
+      equationCalc.nums.splice(signIndex, 1);
+      equationCalc.sign.splice(signIndex, 1);
+
+      signIndex = equationCalc.sign.findIndex(
+        (element) => element === "x" || element === "÷"
+      );
+      if (signIndex === -1) return;
+      handleMultiplicationsAndDivisions(signIndex, equationCalc);
+    }
+  };
+
+  const trasformToPercentage = (mainValue: number, percentageStr: string) => {
+    let percentageValue = parseFloat(percentageStr);
+    percentageValue = (percentageValue / 100) * mainValue;
+    percentageStr = percentageValue.toString();
+    return percentageStr;
+  };
+
   return (
-    <button onClick={updateCalculationState} className="text-3xl">
-      {input.toString()}
-    </button>
+    <li
```

Comment:
[suggestion] Wrapping the PadBtn with a `<li>`works, but what if you need to use this button outside a list? I know you're only using it within a list, but when you build a component you should try to make it as reusable and decoupled from outside logic as you can. So, you could wrap your `<PadBtn>` with a `<li>` in your `calcluator.tsx` file, on line 26, where you render the list.